src/autowiring/AnySharedPointer.h

```diff
@@ -68,6 +68,8 @@ struct AnySharedPointer {
     return obj && try_assign(obj);
   }
 
+  const std::shared_ptr<void>& as_void(void) const { return m_ptr; }
+
```

Comment:
This looks like a means of having an un-typed container for a pointer without losing type safety.  I can understand wanting this to be able to write simpler template logic elsewhere.  What confuses me here is throwing away your ability to guard type safe access by returning a shared_ptr<void>.  Shouldn't callers need to provide the type of the underlying pointer so it can be checked before being allowed access to the shared pointer?